pkg/sypgp/sypgp.go

```diff
@@ -458,11 +460,16 @@ func (keyring *Handle) genKeyPair(name, comment, email, passphrase string) (*ope
 }
 
 // GenKeyPair generates an PGP key pair and store them in the sypgp home folder
-func (keyring *Handle) GenKeyPair(keyServiceURI string, authToken string) (*openpgp.Entity, error) {
+func (keyring *Handle) GenKeyPair(keyServiceURI, authToken string, bitLen int) (*openpgp.Entity, error) {
 	if err := keyring.PathsCheck(); err != nil {
 		return nil, err
 	}
 
+	// Fail early if the bit length is invalid
+	if bitLen < 1024 || bitLen > 4096 {
+		return nil, ErrInvalidBitLen
```

Comment:
Why having this particular check?

Reply:
Should it not be here? I thought the key length was supposed to be in this range. (maybe could start a bit lower, like 7xx bits)...

Reply:
Maybe, this check should be removed? 
If I am not mistaken, RSA does not restrict key len to be in this range as well as to be a power of two.

Reply:
Agreed. No need to artificially restrict, let the underlying RSA library enforce its conditions.